### PR TITLE
feat(button): Allow custom Spinner for Button

### PR DESCRIPTION
**Pull request by yasincaliskan**
### Related Issue
- https://github.com/Hipo/react-ui-toolkit/issues/75

image

### CODE REVIEW

File: src/button/Button.tsx [hunk 1]
```diff
@@ -54,6 +56,14 @@ function Button({
     "button--is-inactive": isButtonDisabled,
     "button--is-pending": shouldDisplaySpinner
   });
+  const displaySpinner = customSpinner || (
```
Comment (mucahit) [inline src/button/Button.tsx:59]:
```suggestion
  const spinnerContent = customSpinner || (
```

File: src/form/input/file/FileInput.tsx [hunk 2]
```diff
@@ -38,6 +40,9 @@ function FileInput({
   const labelClassName = classNames("file-input__label", customLabelClassName, {
     "file-input__label--is-disabled": isInputDisabled
   });
+  const displaySpinner = customSpinner || (
```
Comment (mucahit) [inline src/form/input/file/FileInput.tsx:43]: Same here

**Review by jamcry — APPROVED** [review]

File: src/button/Button.tsx [hunk 3]
```diff
@@ -54,6 +56,14 @@ function Button({
     "button--is-inactive": isButtonDisabled,
     "button--is-pending": shouldDisplaySpinner
   });
+  const spinnerContent = customSpinner || (
+    <Spinner
+      customClassName={classNames("button__spinner", {
+        [`${customClassName}__spinner`]: customClassName
```
Comment (edizcelik) [inline src/button/Button.tsx:62]: Let's not do this because it may lead to weird classes. For example, if we provide customClassName as `button--small button--primary` then this results in adding `button--small button--primary__spinner`. Let's just use `button__spinner` when we want to override the styles of the default Spinner

File: src/toggle/util/toggleHooks.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+import {ToggleContext} from "../Toggle";
+
+/**
+ * @returns {Object} Current value of ToggleContext
+ */
+function useToggleContext() {
+  return useContext(ToggleContext);
+}
+
+export {useToggleContext};
```
Comment (edizcelik) [inline src/toggle/util/toggleHooks.ts:12]: Why is this included here ? 🤔

Comment (yasincaliskan) [reply src/toggle/util/toggleHooks.ts:12]: Ops! It's really weird. I guess my local branches are confused.

File: src/button/Button.tsx (hunk 3 above)
Comment (yasincaliskan) [reply src/button/Button.tsx:62]: Actually, this [PR](https://github.com/Hipo/react-ui-toolkit/pull/81) includes this solution. If it's OK, I can close this PR.

**Review by mucahit — APPROVED** [review]

**Review by Anlerkan — APPROVED** [review]

**Review by bilgeyucel — APPROVED** [review]

**Review by edizcelik — APPROVED** [review]
